pkg/apis/core/validation/namespacedcloudprofile_test.go

```diff
@@ -0,0 +1,1104 @@
+					namespacedCloudProfile.Spec.Regions = duplicatedZones
+
+					errorList := ValidateNamespacedCloudProfile(namespacedCloudProfile)
+
+					Expect(errorList).To(ConsistOf(
+						PointTo(MatchFields(IgnoreExtras, Fields{
+							"Type":  Equal(field.ErrorTypeDuplicate),
+							"Field": Equal(fmt.Sprintf("spec.regions[0].zones[%d].name", len(duplicatedZones[0].Zones)-1)),
+						}))))
+				})
+
+				It("should forbid invalid label specifications", func() {
+					namespacedCloudProfile.Spec.Regions[0].Labels = map[string]string{
+						"this-is-not-allowed": "?*&!@",
+						"toolongtoolongtoolongtoolongtoolongtoolongtoolongtoolongtoolongtoolongtoolong": "toolongtoolongtoolongtoolongtoolongtoolongtoolongtoolongtoolongtoolongtoolong",
+					}
+
+					errorList := ValidateNamespacedCloudProfile(namespacedCloudProfile)
+
+					Expect(errorList).To(ConsistOf(
+						PointTo(MatchFields(IgnoreExtras, Fields{
+							"Type":  Equal(field.ErrorTypeInvalid),
+							"Field": Equal("spec.regions[0].labels"),
+						})),
+						PointTo(MatchFields(IgnoreExtras, Fields{
+							"Type":  Equal(field.ErrorTypeInvalid),
+							"Field": Equal("spec.regions[0].labels"),
+						})),
+						PointTo(MatchFields(IgnoreExtras, Fields{
+							"Type":  Equal(field.ErrorTypeInvalid),
+							"Field": Equal("spec.regions[0].labels"),
+						})),
+					))
+				})
+			})
+
+			Context("volume types validation", func() {
+				It("should enforce uniqueness of volume type names", func() {
+					namespacedCloudProfile.Spec.VolumeTypes = []core.VolumeType{
+						namespacedCloudProfile.Spec.VolumeTypes[0],
+						namespacedCloudProfile.Spec.VolumeTypes[0],
+					}
+
+					errorList := ValidateNamespacedCloudProfile(namespacedCloudProfile)
+
+					Expect(errorList).To(ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{
+						"Type":  Equal(field.ErrorTypeDuplicate),
+						"Field": Equal("spec.volumeTypes[1].name"),
+					}))))
+				})
+
+				It("should forbid volume types with unsupported property values", func() {
+					namespacedCloudProfile.Spec.VolumeTypes = invalidVolumeTypes
+
+					errorList := ValidateNamespacedCloudProfile(namespacedCloudProfile)
+
+					Expect(errorList).To(ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{
+						"Type":  Equal(field.ErrorTypeRequired),
+						"Field": Equal("spec.volumeTypes[0].name"),
+					})), PointTo(MatchFields(IgnoreExtras, Fields{
+						"Type":  Equal(field.ErrorTypeRequired),
+						"Field": Equal("spec.volumeTypes[0].class"),
+					})), PointTo(MatchFields(IgnoreExtras, Fields{
+						"Type":  Equal(field.ErrorTypeInvalid),
+						"Field": Equal("spec.volumeTypes[0].minSize"),
+					})),
+					))
+				})
+			})
+		})
+	})
+
+	var (
+		cloudProfileNew *core.NamespacedCloudProfile
+		cloudProfileOld *core.NamespacedCloudProfile
+		dateInThePast   = &metav1.Time{Time: time.Now().AddDate(-5, 0, 0)}
+	)
+
+	Describe("#ValidateNamespacedCloudProfileSpecUpdate", func() {
```

Comment:
`ValidateNamespacedCloudProfileSpecUpdate` function only checks immutability of the parent.
```suggestion
	Describe("#ValidateNamespacedCloudProfileUpdate", func() {
```